packages/pillarbox-playlist/src/pillarbox-playlist.js

```diff
@@ -0,0 +1,292 @@
+   * the main `ended` handler, ensuring that additional logic can be executed or
+   * making it easier to detach the event listener later.
+   *
+   * @private
+   */
+  onEnded_ = () => this.handleEnded();
+
+  /**
+   * Creates an instance of a pillarbox playlist.
+   *
+   * @param {import('@srgssr/pillarbox-web').Player} player The player instance.
+   * @param {Object} options Configuration options for the plugin.
+   */
+  constructor(player, options) {
+    super(player);
+    if (options.playlist && options.playlist.length) {
+      player.ready(() => {
+        this.load(...options.playlist);
+      });
+    }
+    this.autoadvance = !!options.autoadvance;
+    this.repeat = !!options.repeat;
+    this.player.on('ended', this.onEnded_);
+  }
+
+  dispose() {
+    this.player.off('ended', this.onEnded_);
+  }
+
+  /**
+   * Loads a playlist into the player. This method will load the first element
+   * in the playlist. Use it to initialize the playlist.
+   *
+   * @param {...PlaylistItem} items The playlist items to load.
+   */
+  load(...items) {
+    this.items_ = items;
+    this.select(0);
+  }
+
+  /**
+   * Adds one or more items at the end of the playlist. This method will not
+   * load any of the elements. Use it to add items while the playlist is
+   * running.
+   *
+   * @param {...PlaylistItem} items the items to add to the playlist.
+   */
+  push(...items) {
+    this.items_.push(...items);
+  }
+
+  /**
+   * Modifies the contents of the playlist by removing or replacing existing
+   * elements and/or adding new elements.
+   *
+   * The method also adjusts currentIndex accordingly if items are added or
+   * removed in such a way that it affects the currentIndex.
+   *
+   * If the current item is deleted then the currentIndex becomes -1, the
+   * current element will continue playing but the next element will be the
+   * first element in the playlist.
+   *
+   * @param {number} start The zero-based location in the array from which to
+   *                       start removing elements.
+   * @param {number} deleteCount The number of elements to remove.
+   * @param {...PlaylistItem} items The items to add to the playlist.
+   *
+   * @return {PlaylistItem[]} An array containing the deleted elements.
+   */
+  splice(start, deleteCount, ...items) {
+    const itemsAddedCount = items.length;
+    const effectiveDeleteCount =
+      Math.min(deleteCount, this.items_.length - start);
+
+    // Adjust currentIndex for items being deleted
+    if (this.currentIndex_ >= start) {
+      if (this.currentIndex_ < start + effectiveDeleteCount) {
+        // Current item is removed, set currentIndex to -1
+        this.currentIndex_ = -1;
+      } else {
+        // Adjust currentIndex based on the net items added/removed
+        this.currentIndex_ =
+          this.currentIndex_ - effectiveDeleteCount + itemsAddedCount;
+      }
+    }
+
+    return this.items_.splice(start, deleteCount, ...items);
```

Comment:
Could we do something like the code below to avoid nested `if` ?
```suggestion
    if (this.currentIndex_ < start) {
      return this.items_.splice(start, deleteCount, ...items);
    }

    // Adjust currentIndex for items being deleted
    if (this.currentIndex_ < start + effectiveDeleteCount) {
      // Current item is removed, set currentIndex to -1
      this.currentIndex_ = -1;
    } else {
      // Adjust currentIndex based on the net items added/removed
      this.currentIndex_ =
        this.currentIndex_ - effectiveDeleteCount + itemsAddedCount;
    }

    return this.items_.splice(start, deleteCount, ...items);
```